src/titiler/multidim/reader.py

```diff
@@ -1,24 +1,242 @@
 """XarrayReader"""
 
 import pickle
-from typing import Any, List, Optional
+import os
+from typing import Any, List, Optional, Dict
+from urllib.parse import urlparse
+
 
 import attr
+import xarray as xr
 
 from titiler.multidim.redis_pool import get_redis
 from titiler.multidim.settings import ApiSettings
 from titiler.xarray.io import Reader, get_variable, xarray_open_dataset
 
+try:
+    import icechunk
+except ImportError:  # pragma: nocover
+    icechunk = None  # type: ignore
+
+try:
+    import fsspec
+except ImportError:  # pragma: nocover
+    fsspec = None  # type: ignore
+
+try:
+    import obstore
+except ImportError:  # pragma: nocover
+    obstore = None  # type: ignore
+
+try:
+    import h5netcdf
+except ImportError:  # pragma: nocover
+    h5netcdf = None  # type: ignore
+
+try:
+    import zarr
+except ImportError:  # pragma: nocover
+    zarr = None  # type: ignore
+
 api_settings = ApiSettings()
 cache_client = get_redis()
 
 
+def opener_icechunk(
+    src_path: str,
+    group: Optional[Any] = None,
+    decode_times: bool = True,
+    authorize_virtual_chunk_access: Optional[Dict[str, Dict[str, Any]]] = None,
+) -> xr.Dataset:
+    """Open an IceChunk dataset using xarray."""
+    assert icechunk is not None, "'icechunk' must be installed to read icechunk dataset"
+
+    # TODO: This will likely be repeated across openers. Can we somehow handle this in the Reader Class?
+    parsed = urlparse(src_path)
+    protocol = parsed.scheme or "file"
+
+    authorize_virtual_chunk_access = authorize_virtual_chunk_access or {}
+
+    print(f"DEBUG: authorize_virtual_chunk_access = {authorize_virtual_chunk_access}")
+    if protocol == "file":
+        storage = icechunk.local_filesystem_storage(src_path)
+    elif protocol == "s3":
+        storage = icechunk.s3_storage(
+            bucket=parsed.netloc,
+            prefix=parsed.path.lstrip(
+                "/"
+            ),  # remove leading slash, this is an annoying mismatch between icechunk and urlparse
+            from_env=True,  # we always assume that we can get credentials from env vars or IAM role for the store itself?
+        )
+    else:
+        raise NotImplementedError(
+            f"icechunk storage for protocol {protocol} is not implemented"
+        )
+    # # I think it would be more elegant to get the virtual chunk containers and compare against authorized
+    # # containers from settings but that might be slow?
+    # config = icechunk.Repository.fetch_config(storage=storage)
+    # vchunk_containers = config.virtual_chunk_containers.keys()
+    # container_credentials = icechunk.containers_credentials(
+    #     {k: icechunk.s3_credentials(from_env=True) for k in vchunk_containers}
+    # )
+    # Only create credentials object if there are credentials to authorize
+    # Empty credentials dict causes PyString conversion errors in icechunk
```

Comment:
```suggestion
    # TODO: I think it would be more elegant to get the virtual chunk containers and compare against authorized containers from settings but that might be slowing things down. Leaving this for later.

```